pkg/snmp/snmp_test.go

```diff
@@ -160,16 +160,128 @@ snmp_listener:
+     - network: 127.0.0.1/30
+     - network: 127.0.0.2/30
+       collect_device_metadata: true
+     - network: 127.0.0.3/30
+       collect_device_metadata: false
+`))
+	assert.NoError(t, err)
+
+	conf, err := NewListenerConfig()
+	assert.NoError(t, err)
+
+	assert.Equal(t, "127.0.0.1/30", conf.Configs[0].Network)
+	assert.Equal(t, true, conf.Configs[0].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.2/30", conf.Configs[1].Network)
+	assert.Equal(t, true, conf.Configs[1].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.3/30", conf.Configs[2].Network)
+	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
+
+	// collect_device_metadata: false
+	err = config.Datadog.ReadConfig(strings.NewReader(`
+network_devices:
+  snmp_listener:
+    collect_device_metadata: false
+    configs:
+     - network: 127.0.0.1/30
+     - network: 127.0.0.2/30
+       collect_device_metadata: true
+     - network: 127.0.0.3/30
+       collect_device_metadata: false
+`))
+	assert.NoError(t, err)
+
+	conf, err = NewListenerConfig()
+	assert.NoError(t, err)
+
+	assert.Equal(t, "127.0.0.1/30", conf.Configs[0].Network)
+	assert.Equal(t, false, conf.Configs[0].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.2/30", conf.Configs[1].Network)
+	assert.Equal(t, true, conf.Configs[1].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.3/30", conf.Configs[2].Network)
+	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
+
+	// collect_device_metadata: true
+	err = config.Datadog.ReadConfig(strings.NewReader(`
+network_devices:
+  snmp_listener:
+    collect_device_metadata: true
+    configs:
+     - network: 127.0.0.1/30
+     - network: 127.0.0.2/30
+       collect_device_metadata: true
+     - network: 127.0.0.3/30
+       collect_device_metadata: false
+`))
+	assert.NoError(t, err)
+
+	conf, err = NewListenerConfig()
+	assert.NoError(t, err)
+
+	assert.Equal(t, "127.0.0.1/30", conf.Configs[0].Network)
+	assert.Equal(t, true, conf.Configs[0].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.2/30", conf.Configs[1].Network)
+	assert.Equal(t, true, conf.Configs[1].CollectDeviceMetadata)
+	assert.Equal(t, "127.0.0.3/30", conf.Configs[2].Network)
+	assert.Equal(t, false, conf.Configs[2].CollectDeviceMetadata)
+}
+
+func TestBothListenersConfig(t *testing.T) {
+	config.Datadog.SetConfigType("yaml")
+
+	// default collect_device_metadata should be true
 	err := config.Datadog.ReadConfig(strings.NewReader(`
 snmp_listener:
+  collect_device_metadata: true
   configs:
-   - network: 127.1.0.0/30
-   - network: 127.2.0.0/30
-     loader: core
-   - network: 127.3.0.0/30
-     loader: python
+   - network: 127.0.0.1/30
+   - network: 127.0.0.2/30
+     collect_device_metadata: true
+   - network: 127.0.0.3/30
+     collect_device_metadata: false
+network_devices:
+  snmp_listener:
+    collect_device_metadata: false
+    configs:
+     - network: 127.0.0.1/30
+     - network: 127.0.0.2/30
+       collect_device_metadata: false
+     - network: 127.0.0.3/30
```

Comment:
Let's test different subnets between `snmp_listener` and `network_devices` to ensure the correct one is used

Reply:
updated in [this commit](https://github.com/DataDog/datadog-agent/pull/24644/commits/e16b349fef49ccfe769bb2d561a140e1a4489d4f)